File: scala/2109.adding-spaces-to-a-string.scala
/*
 * @lc app=leetcode id=2109 lang=scala
 *
 * [2109] Adding Spaces to a String
 */

// @lc code=start
object Solution {
    def addSpaces(s: String, spaces: Array[Int]): String = {
        val sb = new StringBuilder
        var prev = 0
        for (space <- spaces) {
            sb.append(s.substring(prev, space)). append(' ')
            prev = space
        }
        sb.append(s.substring(prev))
        sb.toString
    }
}
// @lc code=end

File: python/2109.adding-spaces-to-a-string.py
#
# @lc app=leetcode id=2109 lang=python3
#
# [2109] Adding Spaces to a String
#

# @lc code=start
class Solution:
    def addSpaces(self, s: str, spaces: List[int]) -> str:
        m, n = len(s), len(spaces)
        if n == 0:
            return s
        res = []
        spaces = [0] + spaces + [m]

        for i in range(1, len(spaces)):
            res.append(s[spaces[i-1]:spaces[i]])
            if i < len(spaces) - 1:
                res.append(' ')
        return ''.join(res)
# @lc code=end

